### ISSUE TITLE
DirectionalLight3D acting weird with GPUParticles3D Meshes

### ISSUE THREAD

**Issue by Mikeysax** [godotengine/godot#67599]
### Godot version

4.0beta3

### System information

PopOS 22.04, 5.19.0-76051900-generic kernel, Ryzen 6800h, Radeon 6700m, amdgpu, Vulkan

### Issue description

When the angle of the DirectionalLight3D is at specific angles it makes the meshes from the GPUParticles3D look odd.

I'm not sure if this is a bug or how it's currently setup or if this is a actual known problem currently:

(Video would not upload due to size but here is are screenshots)

![Screenshot from 2022-10-18 19-26-30](https://user-images.githubusercontent.com/15805689/196567003-22db6b24-a328-47a3-8e52-20345e6bb5e1.png)

![Screenshot from 2022-10-18 20-00-43](https://user-images.githubusercontent.com/15805689/196567005-67d546e5-af1b-462e-8380-122926ecc677.png)

![Screenshot from 2022-10-18 20-00-57](https://user-images.githubusercontent.com/15805689/196567006-95d24c76-b479-4671-b9a5-21bf8201d6a3.png)

![Screenshot from 2022-10-18 20-01-06](https://user-images.githubusercontent.com/15805689/196567007-0f1f863a-6080-4325-b2d6-f3a109845524.png)

![Screenshot from 2022-10-18 20-01-31](https://user-images.githubusercontent.com/15805689/196567008-37818c02-60bc-4cdc-a851-f9bbfb2b5c14.png)

Depending on the angle of the light, it changes the lighting of the grass particles in a very drastic manner. It's not smooth and very abrupt.

For contrast, here is what it looks like during the day (this is not unshaded):

![image](https://user-images.githubusercontent.com/15805689/196567265-28de59d7-5ba7-45e5-ae32-e94acc236d55.png)

![image](https://user-images.githubusercontent.com/15805689/196567326-21ff8f78-61eb-42b8-968a-1b951520e82d.png)

It looks fine at some angles but is odd how bright it is. I've tested with a standard DirectionalLight3D without any shaders, and it still makes the grass particles incredibly bright. 

Is this potentially an issue with how particles update lighting information versus if I used a multi-mesh or is this an actual bug?

### Steps to reproduce

Use a DirectionalLight3D with GPUParticle3D meshes at different angles of directional light 3d

**Comment by clayjohn** [comment]
Could you please upload an MRP so that contributors can attempt to reproduce the issue you are facing?

Based on the screenshots I would guess that the normals in the mesh are improperly created. You can take a look at the normals by clicking on the "Perspective" menu and sleecting "Display Advanced... > normal buffer". Particles are rendered the exact same way as Multimesh, i.e. using the material that is assigned to them.

**Comment by Mikeysax** [comment]
> Could you please upload an MRP so that contributors can attempt to reproduce the issue you are facing?
> 
> Based on the screenshots I would guess that the normals in the mesh are improperly created. You can take a look at the normals by clicking on the "Perspective" menu and sleecting "Display Advanced... > normal buffer". Particles are rendered the exact same way as Multimesh, i.e. using the material that is assigned to them.

I see; thanks for this. Here is a screenshot:

![image](https://user-images.githubusercontent.com/15805689/202432265-5e9b9710-8a7a-4cd4-bb9a-ee2aa7c3fbbf.png)

I'm assuming this isn't an issue @clayjohn , given the normals are not correct. Please close the issue if this is the case

**Comment by Mikeysax** [comment]
@clayjohn So, I ended up figuring out what it was. The normals were correct but the `cull_disabled` render_mode flag caused this issue. Once I removed this, it fixed the issue.

In order for `cull_disabled` to be there and not have a double sided mesh with normals, I needed to do the following:

Add this to the `vertex` shader:

```
NORMAL = vec3(0.0, 1.0, 0.0);
```

Add this to the `fragment` shader:

```
NORMAL *= FRONT_FACING ? 1.0 : -1.0;
```
